schema.go

```diff
@@ -769,20 +546,20 @@ func getTypeKind(t reflect.Type) (string, bool) {
 	return typeName, isEnum
 }
 
-func serializeTypeRef(t reflect.Type, info serializationInfo) (*schema.TypeSpec, error) {
+func (info serializationInfo) serializeTypeAny(t reflect.Type) (*schema.TypeSpec, error) {
 	typeKind, enum := getTypeKind(t)
 	if enum {
-		enumSpec, err := serializeEnum(t, info)
+		enumSpec, err := info.serializeTypeEnum(t)
```

Comment:
`serializeEnumType` is easier to read then `serializeTypeEnum`. Especially since other versions of this function put the type kind in the middle, see `serializeReferenceType`.

Reply:
There is already a function named `serializeEnumType` which returns a `schema.ComplexTypeSpec`. Maybe we could rename all functions which return `schema.TypeSpec` to `serializeAny`, `serializeReference`, `serializeEnum`?